### PR TITLE
gh-2244 FederatedStore ChangeGraphId

### PR DESCRIPTION
**Pull request by GCHQDev404**

### CODE REVIEW

File: store-implementation/federated-store/src/main/java/uk/gov/gchq/gaffer/federatedstore/operation/ChangeGraphId.java [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+import uk.gov.gchq.koryphe.Summary;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import static uk.gov.gchq.gaffer.federatedstore.FederatedStoreConstants.KEY_OPERATION_OPTIONS_GRAPH_IDS;
+
+
+@JsonPropertyOrder(value = {"class", "graphId", "newGraphId"}, alphabetic = true)
+@Since("1.10.6")
```
Comment (d47853) [inline store-implementation/federated-store/src/main/java/uk/gov/gchq/gaffer/federatedstore/operation/ChangeGraphId.java]: Needs updating to 1.12.0

File: store-implementation/federated-store/src/main/java/uk/gov/gchq/gaffer/federatedstore/operation/ChangeGraphId.java [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+@Summary("Changes the protection used for accessing graphs")
```
Comment (d47853) [inline store-implementation/federated-store/src/main/java/uk/gov/gchq/gaffer/federatedstore/operation/ChangeGraphId.java]:
```suggestion
@Summary("Changes the Id of a graph")
```

File: store-implementation/federated-store/src/main/java/uk/gov/gchq/gaffer/federatedstore/operation/ChangeGraphId.java [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+@JsonInclude(Include.NON_DEFAULT)
+public class ChangeGraphId implements Output<Boolean> {
+    @Required
+    private String graphId;
+    private String newGraphId;
+    private Map<String, String> options = new HashMap<>();
+
+    public ChangeGraphId() {
+        addOption(KEY_OPERATION_OPTIONS_GRAPH_IDS, "");
```
Comment (d47853) [inline store-implementation/federated-store/src/main/java/uk/gov/gchq/gaffer/federatedstore/operation/ChangeGraphId.java]: Is this done elsewhere? Seems like there should be a removeOption in the Operation interface?

Comment (GCHQDev404) [reply store-implementation/federated-store/src/main/java/uk/gov/gchq/gaffer/federatedstore/operation/ChangeGraphId.java]: It replaced the need to stick `""` as default `getCleanStrings(config.get(KEY_OPERATION_OPTIONS_GRAPH_IDS));`

but happy to change it.

Comment (GCHQDev404) [reply store-implementation/federated-store/src/main/java/uk/gov/gchq/gaffer/federatedstore/operation/ChangeGraphId.java]: I don't think there is a remove option in the interface.

Comment (d47853) [reply store-implementation/federated-store/src/main/java/uk/gov/gchq/gaffer/federatedstore/operation/ChangeGraphId.java]: Could you use getOptions().remove()?

Comment (GCHQDev404) [reply store-implementation/federated-store/src/main/java/uk/gov/gchq/gaffer/federatedstore/operation/ChangeGraphId.java]: Yes :D I forgot I had direct access to the options object. I was working through the Operations interface.

Comment (GCHQDev404) [reply store-implementation/federated-store/src/main/java/uk/gov/gchq/gaffer/federatedstore/operation/ChangeGraphId.java]: Null and "" provide different behaviours.
FederatedStore#getStreams this was a change by @p013570 
This is done is 6 other classes.

**Pull request merged**